docs/resource-schemas.md

```diff
@@ -321,10 +311,92 @@ have the following values:
 | `cross-platform` | Runs natively on iOS, Android, and/or desktop. |
 | `web`            | Runs on the web.                               |
 
+## Design kit schema
+
+Design kits contain elements, components, guidelines and wireframes for various design tools as the
+design counterpart to coded assets. Design kits are indexed in the
+[carbon-platform](https://github.com/carbon-design-system/carbon-platform) repository
+[`carbon.yml`](https://github.com/carbon-design-system/carbon-platform/blob/main/services/web-app/data/carbon.yml)
+data file.
+
```

Comment:
Does it make more sense to attach one or more design kits to each asset as opposed to a library? If we index components along with the design kits that contain them, we can ascertain the design kits that apply at the library level as the group of all design kits we discovered across the assets _within_ that library.

Otherwise, it feels like we might be missing the connection between an asset and the design kits in which its ui/ux design is contained.

Reply:
Thinking in terms of graphql, it feels useful to be able to do a query like:

```gql
query {
  libraries(id=carbon-react) {
    assets(name=accordion) {
      id
      name
      designKits {
        name
      }
    }
  }
}
```

As a way to figure out the design kit(s) that contains a particular component. If we don't have design kit association at the asset level, we can't do that.

To close the loop, to fulfill the query to get the design kits for a library, it would look like:

```gql
query {
  libraries(id=carbon-react) {
    designKits {
      name
    }
  }
}
```

and wouldn't require anything beyond having assets associated with design kits in the indexes.

And going the "other" other direction, this would allow us to do:

```gql
query {
  designKits {
    assets(id=accordion) {
      id
      name
      framework
    }
  }
}
```

as a way to see what assets exist in each design kit.

Reply:
Thanks for providing those example queries. It really helps explain the question. Let's talk through this one with design.

I see how we could add design kit associations with assets too - not just libraries, but that'd be much more tedious to index. For example, there's a nice simplicity to just providing this once at the library level and not requiring it for every asset in that library:

```yml
  designKits:
    - carbon-white-sketch
    - carbon-g10-sketch
    - carbon-g90-sketch
    - carbon-g100-sketch
    - carbon-white-figma
    - carbon-g10-figma
    - carbon-g90-figma
    - carbon-g100-figma
    - carbon-white-adobe-xd
    - carbon-g10-adobe-xd
    - carbon-g90-adobe-xd
    - carbon-g100-adobe-xd
    - carbon-g10-axure
    - carbon-mid-fi-sketch
    - carbon-wireframe-invision-freehand
```

Where we need @carbon-design-system/carbon-platform-designers input - we only have library-specific design kits page at the library level, and no page or section for this content at the per-asset basis:

https://next.carbondesignsystem.com/assets/carbon-react/latest/design-kits

Since every asset belongs to a library, the thinking here is that we can just provide design kit associations at the library level to keep it simple.